src/TLuaInterpreter.cpp

```diff
@@ -16965,3 +16977,63 @@ int TLuaInterpreter::showNotification(lua_State* L)
     mudlet::self()->mTrayIcon.showMessage(title, text, mudlet::self()->mTrayIcon.icon(), notificationExpirationTime);
     return 0;
 }
+
+int TLuaInterpreter::exportJsonMap(lua_State* L)
+{
+    Host* pHost = &getHostFromLua(L);
+    if (!pHost || !pHost->mpMap || !pHost->mpMap->mpMapper || !pHost->mpMap->mpMapper->mp2dMap) {
+        lua_pushnil(L);
+        lua_pushstring(L, "exportJsonMap: no map present or loaded!");
+        return 2;
+    }
+
+    if (!lua_isstring(L, 1)) {
+        lua_pushfstring(L, "exportJsonMap: bad argument #1 type (export pathFileName as string expected, got %s!)", luaL_typename(L, 1));
+        return lua_error(L);
+    }
+    QString dest = lua_tostring(L, 1);
+    if (dest.isEmpty()) {
+        lua_pushnil(L);
+        lua_pushstring(L, "a non-empty path and file name to write to must be provided");
+        return 2;
```

Comment:
```suggestion
        return warnArgumentValue(L, __func__, QStringLiteral("a non-empty path and file name to write to must be provided"));
```

Reply:
Fair enough - but this started life before you got to work in there... :tools: